components/homme/src/share/prim_driver_base.F90

```diff
@@ -1273,11 +1275,22 @@ subroutine prim_step_flexible(hybrid, elem, nets, nete, dt, tl, hvcoord, compute
     real(kind=real_kind), intent(in)    :: dt       ! "timestep dependent" timestep
     type(TimeLevel_t),    intent(inout) :: tl
     logical,              intent(in)    :: compute_diagnostics
+    logical,              intent(in)    :: single_column
 
     real(kind=real_kind) :: dt_q, dt_remap, dp(np,np,nlev)
-    integer :: ie, q, k, n, n0_qdp, np1_qdp
+    integer :: ie, q, k, n, n0_qdp, np1_qdp, nets_in, nete_in
     logical :: compute_diagnostics_it, apply_forcing
 
+    logical :: scm_update_T
+
+    if (single_column) then
+      nets_in=1
```

Comment:
As a style point, you might consider using nets_in, nete_in as the arguments to the function and make nets, nete the local variables. That way nets, nete get used throughout the definition rather than nets_in, nete_in.